nixos/modules/hardware/video/high-dpi.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, pkgs, config, ...}:
+with lib;
+
+{
+  options = {
+    hardware.video.high-dpi.enable = mkEnableOption "Font/DPI configuration optimized for High-DPI displays";
+  };
+  config = mkIf config.hardware.video.high-dpi.enable {
+    console.font = mkDefault
+      "${pkgs.terminus_font}/share/consolefonts/ter-u28n.psf.gz";
+
+    # Needed when typing in passwords for full disk encryption
+    console.earlySetup = mkDefault true;
+    boot.loader.systemd-boot.consoleMode = mkDefault "max";
+
+    services.xserver.dpi = mkDefault 196;
+    fonts.fontconfig.dpi = mkDefault 196;
```

Comment:
So I assume that everything starting from here should be only set, if x11 is enabled, or will wayland not be bothered by it?

Reply:
@Mic92 Not bothered as I looked, and these should help XWayland programs

Reply:
Generally X11 is enabled even for pure-Wayland configurations (NixOS conflates "graphical environment" with "Xorg enabled" currently), so it's impossible to make configurations conditional on not using Wayland.